source/common/http/filter/buffer_filter.cc

```diff
@@ -54,13 +54,14 @@ BufferFilterStats BufferFilter::generateStats(const std::string& prefix, Stats::
   return {ALL_BUFFER_FILTER_STATS(POOL_COUNTER_PREFIX(scope, final_prefix))};
 }
 
-void BufferFilter::onDestroy() { resetInternalState(); }
+void BufferFilter::onDestroy() {
+  resetInternalState();
+  stream_destroyed_ = true;
```

Comment:
It feels almost like we want a trait here, but alas C++ and multiple inheritance.

Reply:
Yeah agreed some kind of base class would be better but not sure if it's worth trying to figure that out here.